string_grouper/string_grouper.py

```diff
@@ -128,6 +134,8 @@ class StringGrouperConfig(NamedTuple):
     min_similarity: float = DEFAULT_MIN_SIMILARITY
     number_of_processes: int = DEFAULT_N_PROCESSES
     ignore_case: bool = DEFAULT_IGNORE_CASE
+    drop: bool = DEFAULT_DROP_INDEX
```

Comment:
I had thought similarly.  But I found myself pulled in opposite directions, since pandas itself uses the keyword argument `drop` for an almost identical purpose as we do.  So I thought for consistency with pandas it should be `drop`.  But you perhaps do not concur?

Reply:
On second thought, I now realize that in pandas the kwarg `drop` is only used in calls of `reset_index` where it is clear that it is the index that is being dropped, unlike in StringGrouper.  So I refactored it to `ignore_index`.